src/middlewared/middlewared/plugins/jail.py

```diff
@@ -822,33 +826,38 @@ def exec(self, jail, command, options):
     @job(lock=lambda args: f"jail_update:{args[-1]}")
     def update_to_latest_patch(self, job, jail):
         """Updates specified jail to latest patch level."""
+        job.set_progress(0, f'Updating {jail}')
+        msg_queue = deque(maxlen=10)
 
-        uuid, path, _ = self.check_jail_existence(jail)
-        status, jid = IOCList.list_get_jid(uuid)
-        conf = IOCJson(path).json_load()
-
-        # Sometimes if they don't have an existing patch level, this
-        # becomes 11.1 instead of 11.1-RELEASE
-        _release = conf["release"].rsplit("-", 1)[0]
-        release = _release if "-RELEASE" in _release else conf["release"]
-
-        started = False
-
-        if conf["type"] == "jail":
-            if not status:
-                self.start(jail)
-                started = True
-        else:
-            return False
-
-        if conf["basejail"] != "yes":
-            IOCFetch(release).fetch_update(True, uuid)
-        else:
-            # Basejails only need their base RELEASE updated
-            IOCFetch(release).fetch_update()
+        def progress_callback(content, exception):
+            msg = content['message'].strip('\n')
+            msg_queue.append(msg)
+            final_msg = '\n'.join(msg_queue)
+
+            if 'Inspecting system... done' in msg:
+                job.set_progress(20)
+            elif 'Preparing to download files... done.' in msg:
+                job.set_progress(50)
+            elif 'Applying patches... done.' in msg:
+                job.set_progress(75)
+            elif 'Installing updates... done.' in msg:
+                job.set_progress(90)
+            elif f'{jail} has been updated successfully' in msg:
+                job.set_progress(100)
+
+            job.set_progress(None, description=final_msg)
+
+        _, path, iocage = self.check_jail_existence(
+            jail,
+            callback=progress_callback
+        )
+        jail_type = IOCJson(path).json_load()['type']
+        iocage.update()
 
-        if started:
-            self.stop(jail)
+        if jail_type == 'pluginv2':
```

Comment:
Also is this true for `plugin` jail_type ?

Reply:
No, those are old and don't have many of the things we expect.